File: Learning Android App/Search Button/app/src/main/java/com/example/searchbutton/MainActivity.java
package com.example.searchbutton;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SearchView;
import androidx.recyclerview.widget.LinearLayoutManager;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.Menu;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;

import com.example.searchbutton.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());


    }

    @SuppressLint("ResourceType")
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {




        return super.onCreateOptionsMenu(menu);
    }
}

File: Learning Android App/retrofitMVVM/app/src/main/java/com/example/retrofitmvvm/MainActivity.kt
package com.example.retrofitmvvm

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.CountDownTimer
import android.util.Log
import android.view.View
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.example.QuizData
import com.example.retrofitmvvm.Repository.Retrofit.GetDataAPI
import com.example.retrofitmvvm.Repository.Retrofit.RetrofitResponse
import com.example.retrofitmvvm.ViewModel.QuizViewModel
import com.example.retrofitmvvm.databinding.ActivityMainBinding
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import retrofit2.Call
import retrofit2.Callback
import retrofit2.HttpException
import retrofit2.Response
import java.io.IOException
import java.util.*

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private lateinit var viewModel: QuizViewModel


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)


        viewModel = ViewModelProvider(this)[QuizViewModel::class.java]

        viewModel.getQuizLiveDataRepo()?.observe(this, androidx.lifecycle.Observer {


            var quiz = it as QuizData
            var adapter = QuizAdapter()
            adapter.setQuizData(quiz.result.questions)
            binding.quizRecyclerView.adapter = adapter
            adapter.notifyDataSetChanged()
        })



        binding.quizRecyclerView.layoutManager = LinearLayoutManager(this)
    }

}